pkg/blockbuilder/schedulerpb/client_test.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package schedulerpb
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	grpc "google.golang.org/grpc"
+
+	"github.com/grafana/mimir/pkg/util/test"
+)
+
+func TestGetJob(t *testing.T) {
+	sched := &mockSchedulerClient{}
+	sched.key = JobKey{Id: "foo/983/585", Epoch: 4523}
+	sched.spec = JobSpec{Topic: "foo", Partition: 983, StartOffset: 585}
+
+	cli := NewSchedulerClient("worker1", sched, test.NewTestingLogger(t), 0, 0)
+	ctx := context.Background()
+	key, spec, err := cli.GetJob(ctx)
+
+	require.NoError(t, err)
+	require.Equal(t, sched.key, key)
+	require.Equal(t, sched.spec, spec)
+	require.Equal(t, 1, sched.assignCalls)
+}
+
+func TestCompleteJob(t *testing.T) {
+	sched := &mockSchedulerClient{}
+	sched.key = JobKey{Id: "foo/983/585", Epoch: 4523}
+	sched.spec = JobSpec{Topic: "foo", Partition: 983, StartOffset: 585}
+
+	cli := NewSchedulerClient("worker1", sched, test.NewTestingLogger(t), 0, 64*time.Hour).(*schedulerClient)
+	ctx := context.Background()
+	key, _, err := cli.GetJob(ctx)
+	require.NoError(t, err)
+
+	require.NoError(t, cli.CompleteJob(key))
+	require.True(t, cli.jobs[key].complete)
+	require.True(t, cli.jobs[key].forgetTime.After(time.Now()))
+
+	require.ErrorContains(t, cli.CompleteJob(JobKey{Id: "erroneous id"}), "not found")
+}
+
+func TestSendUpdates(t *testing.T) {
```

Comment:
Let's test two more things:
1. `sendUpdates` updating more than one job.
2. Verifying that the right details were sent in the UpdateJob call in addition to checking if UpdateJob was called. This can also be merged with `TestCompleteJob` by having one complete and one incomplete job.